### PR TITLE
Strict-type `Dependabot::Docker::Version`

### PR DESCRIPTION
**Pull request by raj-meka**
…ersion ruby file.

### CODE REVIEW

**Review by abdulapopoola — APPROVED** [review]

File: docker/lib/dependabot/docker/version.rb [hunk 1]
```diff
@@ -13,54 +14,65 @@ module Docker
     # for a description of Java versions.
     #
     class Version < Dependabot::Version
+      extend T::Sig
       # The regex has limits for the 0,255 and 1,255 repetitions to avoid infinite limits which makes codeql angry.
       # A docker image cannot be longer than 255 characters anyways.
       DOCKER_VERSION_REGEX = /^(?<prefix>[a-z._\-]{0,255})[_\-v]?(?<version>.{1,255})$/
 
+      sig { override.params(version: T.nilable(T.any(String, Integer, Gem::Version))).void }
```
Comment (JamieMagee) [inline docker/lib/dependabot/docker/version.rb:22]: I had completely forgotten that we had defined a [type alias][1] for this big type. It's kinda long, and this makes it easier: 

```suggestion
      sig { override.params(version: VersionParameter).void }
```

[1]: https://sorbet.org/docs/type-aliases

File: docker/lib/dependabot/docker/version.rb [hunk 2]
```diff
@@ -13,54 +14,65 @@ module Docker
...
+        @release_part = T.let(Dependabot::Version.new(release_part.tr("-", ".")), Dependabot::Version)
+        @update_part = T.let(
+          Dependabot::Version.new(update_part&.start_with?(/[0-9]/) ? update_part : 0),
+          Dependabot::Version
+        )
 
         super(@release_part)
       end
 
+      sig { override.params(version: T.nilable(T.any(String, Integer, Gem::Version))).returns(T::Boolean) }
```
Comment (JamieMagee) [inline docker/lib/dependabot/docker/version.rb:40]: Same as above:

```suggestion
      sig { override.params(version: VersionParameter).returns(T::Boolean) }
```

**Review by JamieMagee — APPROVED** [review]
Approved pending comments

**Comment by abdulapopoola** [comment]
[Parent Issue](https://github.com/github/dependabot-updates/issues/6178)

**Pull request merged**